### ISSUE TITLE
Python Interpreter not matching between Language and "Syntax & Spell Checking" menus

### ISSUE THREAD

**Issue by clokep** [Komodo/KomodoEdit#2067]
### Short Summary

Choosing a custom interpreter under Language > Python does not seem to be reflected in "Syntax & Spell Checking" (at least when using Project settings.

### Steps to Reproduce

1. Create a project.
1. Create a virtual environment
2. Activate the virtual environment
3. `pip install pylint`
2. Go into "Project Settings"
3. Go to "Project Preferences" > "Languages" > "Python"
4. Under "Default Python Interpreter" choose the one in your virtual environment (for me this is something like `/Users/clokep/.virtualenvs/<whatever you called it above>/bin/python`)
5. Under "Additional Python Import Directories" add your site-packages, e.g. `/Users/clokep/.virtualenvs/<whatever you called it above>/lib/python2.7/site-packages`
6. Go to "Project Preferences" > "Syntax & Spell Checking"
9. Choose "Languages" to "Python" , if it isn't already
10. "Do Pylint Checking" will be disabled with a "The current Python instance, /usr/bin/python, doesn't have pylint installed."

### Expected results

I can enable pylint.

### Actual results

Enabling pylint is not available to me.

### Platform Information

*Komodo Edit or IDE?* IDE
*Komodo Version?* Komodo IDE, version 10.1.1, build 89474, platform macosx. Built on Thu Sep 1 15:11:44 2016.
*Operating System (and version)?* Mac OS X 10.11.6

### Additional Information

I didn't see anything useful in the error logs.

I actually have pylint installed globally too, so I'm not really sure why it isn't picking that one up if it is actually using `/usr/bin/python`. Note that it *shouldn't* even be using `/usr/bin/python`, it should use `/usr/local/bin/python` that I installed via homebrew... Komodo preferences don't seem to show what is actually chosen if you leave it at "Find on path" though.

![screen shot 2016-09-29 at 12 57 14 pm](https://cloud.githubusercontent.com/assets/517124/18963773/49547b98-8644-11e6-907c-6c7dfe9f41ca.png)

**Comment by cgchoffman** [comment]
> Note that it shouldn't even be using /usr/bin/python, it should use /usr/local/bin/python that I installed via homebrew

This shouldn't be possible.  Komodo uses `which` (`where` on Win) which returns the first available executable.  There are a number of possible explanations for this given it's Linux.  For example, starting Komodo from the Desktop but testing `which` in a terminal which modifies your path when you open it (`.profile`).  

Having said that though, it should be using what you've set in your project prefs, not global or it defeats the purpose of even having this preference.  This sounds very familiar.  Going to search the tickets.

**Comment by clokep** [comment]
It looks like I *do* modify my path in my `.profile`: `export PATH=/usr/local/bin:/usr/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11/bin:$PATH`. That was an aside though, the main issue is that setting it in the Project preferences doesn't seem to follow through!

Thanks for taking a look. 👍

**Comment by cgchoffman** [comment]
> That was an aside though,

Yep I realize that but you pointed it out as an issue which I was certain it's not.  *Occam's razor* kind of thing :D

**Comment by cgchoffman** [comment]
@clokep can you try setting your project level *language* pref to the Python interpreter that has `pylint` installed and see if the Project > Syntax Checking prefs behave in that scenario?

**Comment by cgchoffman** [comment]
I can't find a related issue in the tickets so we'll just continue with this as a new issue.

**Comment by clokep** [comment]
@cgchoffman It's already set to the proper Python interpreter:

![screen shot 2016-09-29 at 2 14 19 pm](https://cloud.githubusercontent.com/assets/517124/18966414/0e37319e-864f-11e6-9c4a-73f06dfdc210.png)

**Comment by cgchoffman** [comment]
I'm asking for you to test the non-virtualenv interpreter.

**Comment by clokep** [comment]
@cgchoffman I set the "Default Python Interpreter" to one of the ones in the dropdown, which is `/usr/local/bin/python`.

If I deactivate my virtualenv in a terminal and run `which python`, I get `/usr/local/bin/python`. Additionally, running `pylint` works and `which pylint` returns `/usr/local/bin/pylint`.

**Comment by cgchoffman** [comment]
K thank you for checking.  It is possible that it's using the global prefs though.  Sorry to be a pest but could you double checking that when you change the *global pref > lang > Python* to `/usr/bin/python` (the one that doesn't have *pylint*) your project has *pylint* available in the *Project level > Prefs > Syntax Checking*.  Make sure *project level >pref > langs > python* is set to the `/usrr/local/bin/python`.

I'm trying to mimic your env locally but `pip install pylint` just exploded all over my terminal.

**Comment by cgchoffman** [comment]
Ok...I THINK I know what's going on here.  The title of the ticket isn't quite right.

### Trouble:
You can't just point Komodo at the virtualenv (VE) interpreter.  You have to also "activate" the VE (which is why my `VE/Scripts/pip.exe install pylint` blew up).  You tried to make up for this by adding *VE/Lib/site-packages* to the "additional directories" field in the Languages > Python section of your Project prefs.  This doesn't work which I think is the actual bug here.  I'll talk more below.

### Additional Includes

This does appear to be working BUT the prefs dialog isn't including that dir when it checks to see if *pylint* is installed.
NOTES:
- if I open a Python interactive shell with this setup and write `import pylint` it works fine.  
- If I use Tools > Run Command: `which pylint`, this finds nothing.  This matches with Syntax Prefs thinking pylint isn't installed.

### Summary

It looks like when we use the configured Python interpreter to execute a command (ie. `which pylint`) it does *not* add the `additional directories` to the PYTHONPATH in the environment it's running in.  I can't even begin to speculate on why.  I've never looked at this code.  @clokep do you mind if I close this ticket and open a new one with the findings I have?  ie. configured python commands not using configured Additional dirs.

### Workaround

To get this working, from your terminal, "activate" the VE then start Komodo from that terminal.  Everything will work now.  Note that this would be addressed #2068.

**Comment by clokep** [comment]
@cgchoffman Feel free to close this issue and open a new one (please just post the number in here so I can subscribe!) or feel free to change the title of this and morph it into what you need. I can confirm your workaround works! 👍

**Comment by Naatan** [comment]
Nice digging @cgchoffman !

**Comment by cgchoffman** [comment]
Done.  See above.

**Comment by cgchoffman** [comment]
So much going on here.  Sorry, @clokep, I got my mindbrain wrapped in a bit of a knot.  You're report is still valid but I found another issue while looking into it.

So yes, when you set an interpreter in Langs > Python, the syntax checking screen will still show whatever is found on the path first, *NOT* what you have explicitly set in your Language prefs.
